### PR TITLE
  WD-5230 Add homepage contact form

### PR DESCRIPTION
**Pull request by carkod**
## Done

Add contact form for Canonical.com homepage
## QA

- Go to https://canonical-com-1033.demos.haus/ click on "Contact-us"
- Compare against [copy docs](https://docs.google.com/document/d/1saDssfbHdM0pJgJdQmJG7uB6IefXlfCk-c-mCURohB4/edit)
- Check that the form works

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-5230

## Screenshots

[if relevant, include a screenshot]

### CODE REVIEW

File: templates/partners/partial/_homepage-form.html [hunk 1]
```diff
@@ -0,0 +1,89 @@
+<div class="p-modal" style="display: none;" id="executive-summit-modal">
+  <div class="p-modal__dialog" role="dialog" aria-labelledby="modal-title" aria-describedby="modal-description">
+    <header class="p-modal__header" style="display: block; border-bottom: 0;">
+      <button class="p-modal__close" aria-controls="executive-summit-modal" aria-label="Close active modal" style="margin-left: -1rem">Close</button>
+      <h2 class="p-modal__title u-sv1" id="modal-title">Contact Canonical</h2>
+    </header>
+    <div class="js-pagination js-pagination--1">
+      <div class="js-formfield">
+        <h3 class="p-heading--5">Please fill in this form and a member of our team will be in touch soon.</h3>
```
Comment (pedoch) [inline templates/partners/partial/_homepage-form.html:9]: I think the secondary heading should say: "_Fill in this form and we'll be in touch within one working day_." based on what's in the copy doc.

File: templates/base_index.html [hunk 2]
```diff
@@ -119,6 +119,17 @@
...
+        let params = new URLSearchParams(window.location.search);
+        const utm_names = ["campaign", "source", "medium", "content", "term"];
+        for (let i = 0; i < utm_names.length; i++) {
+          var utm_fields = document.getElementsByName("utm_" + utm_names[i]);
+          for (let j = 0; j < utm_fields.length; j++) {
+            if (utm_fields[j]) {
+              utm_fields[j].value = params.get("utm_" + utm_names[i]);
+            }
+          }
+        }
```
Comment (pedoch) [inline templates/base_index.html:132]: I don't see why this needs to be nested for loops, seems to me that the `utm_fields` variable will always contain one field.

I think the code should look something like this:

```
for (let i = 0; i < utm_names.length; i++) {
  let utm_field = document.getElementsByName("utm_" + utm_names[i])[0];

  if (utm_field) {
    utm_field.value = params.get("utm_" + utm_names[i]);
  }
}
```

Unless there's a posibility we'll have duplicate utm fields on a page. I'll leave that up to you to decide.

**Review by pedoch — APPROVED** [review]
LGTM! Just some minor comments

File: templates/base_index.html (hunk 2 above)
Comment (carkod) [reply templates/base_index.html:132]: Yeah I think it does make sense what you say. I believe this was in place to fix in all forms that had duplicated fields, but then that fix would mean one value gets overriden by the other. So you're right, maybe we should remove the second loop.

**Comment by Sophie-32** [comment]
@carkod - I was checking the form on canonical.com, it doesn't match the copydoc from this PR. Let me know if you want to talk it through. I left comments in the copydoc flagging the specific items that are missing.

**Comment by carkod** [comment]
> @carkod - I was checking the form on canonical.com, it doesn't match the copydoc from this PR. Let me know if you want to talk it through. I left comments in the copydoc flagging the specific items that are missing.

mm I just double-checked now, which copy doc are you looking at? It does match https://docs.google.com/document/d/1saDssfbHdM0pJgJdQmJG7uB6IefXlfCk-c-mCURohB4/edit#heading=h.9ewa5gu8qwzo?

**Pull request merged**